torch/distributed/distributed_c10d.py

```diff
@@ -226,6 +229,7 @@ def _get_group_size(group):
     """
     if group is GroupMember.WORLD:
         _check_default_pg()
+        assert _default_pg is not None
```

Comment:
Wouldn't it be a bit cleaner, if _check_default_pg() would return an non-optional PG which one then binds to a local variable?

Reply:
similar to 
```
def _check_default_pg():
  if _default_pg is not None:
    return _default_pg
  else: 
    raise RuntimeError("Default process group is not initialized")
```

Reply:
Good point, fixed by returning ProcessGroup in _check_default_pg(), and assign the return value to a local variable "default_pg".